### PR TITLE
Feat: adding Proof Configuration to Signature

### LINKED ISSUE #34: Signature `ProofOptions` not included
Since I'm running a couple of tests with the MIW and it's VCs, I ran into the following:

I fetch a VC from MIW.
I create a VP with this VC (with MIW)
I verify the VP (and its VCs) with the MIW and it is valid.

I change the VC `proof.created`
I create a VP with this changed VC (with MIW)
I verify the VP (and its VCs) with the MIW and it is valid.

I think the last step should NOT be a valid result!

It looks to me like for creating the signature, only the 'unsecuredDocument' is taken, but not the 'proofOptions'.

Can you please double check this desired behavior? To my understanding it should be the hash from the 'proofOptions' +  the hash from the 'unsecuredDocument' that are concatenated and then used as the data that is signed.

Can we add a test case that verifies the produced VC/VPs with a 3rd party library? Just to make sure compatibility is ok.

--
Matthias Binzer

### PR DESCRIPTION
**Pull request by koptan**
## Description

PR to include the proof configuration in the final hash during the process of creating and verifying the proof.

Therefore, the verification process should fail if any values in the Proof Configuration attributes change.

This PR should fix #34, also it's merge  #38 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by koptan** [comment]
@DominikPinsel @matgnt  @borisrizov-zf @bjoern-arnold, please for your review.

**Comment by matgnt** [comment]
I would leave this to the others to review. I'm not a Java developer and don't know the libraries internal structure.

**Review by DominikPinsel — APPROVED** [review]
I like the idea with the new ProofSignature objects :)

**Comment by borisrizov-zf** [comment]
@koptan you should rebase, because of #44

**Comment by koptan** [comment]
@borisrizov-zf  Rebase is done, you can merge it 👍

File: src/main/java/org/eclipse/tractusx/ssi/examples/Verification.java [hunk 1]
```diff
@@ -67,26 +66,10 @@ public static void verifyJWT(SignedJWT jwt)
...
-   * @throws NoVerificationKeyFoundException
-   * @throws SignatureVerificationException
-   * @throws InvalidPublicKeyFormatException
-   * @throws DidParseException
-   * @throws SignatureParseException
-   * @throws UnsupportedSignatureTypeException
-   * @throws SignatureVerificationFailedException
-   */
-  public static boolean verifyED25519LD(VerifiableCredential verifiableCredential)
+  public static boolean verifyED21559LD(VerifiableCredential verifiableCredential)
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/examples/Verification.java:69]:
```suggestion
  public static boolean verifyED25519LD(VerifiableCredential verifiableCredential)
```

File: src/main/java/org/eclipse/tractusx/ssi/lib/exception/did/DidParseException.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*
  * ******************************************************************************
- * Copyright (c) 2021,2024 Contributors to the Eclipse Foundation
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/exception/did/DidParseException.java:3]: Year should be 2024

File: src/main/java/org/eclipse/tractusx/ssi/lib/exception/json/TransformJsonLdException.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /*
  * ******************************************************************************
- * Copyright (c) 2021,2024 Contributors to the Eclipse Foundation
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/exception/json/TransformJsonLdException.java:3]: Year should be 2024

File: src/main/java/org/eclipse/tractusx/ssi/lib/exception/key/InvalidPrivateKeyFormatException.java [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /*
  * ******************************************************************************
- * Copyright (c) 2021,2024 Contributors to the Eclipse Foundation
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/exception/key/InvalidPrivateKeyFormatException.java:3]: Year should be 2024

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]
Please re-check your PR, the copyright year seems to be wrong on multiple files. Also method documentation has been removed, but it is essential to have documentation to be Maven Central compliant.
I'll re-do the review after these issues have been fixed.

**Comment by borisrizov-zf** [comment]
@koptan Hi, still many more files have the docs deleted, and the copyright year is not 2024. I haven't marked them explicitly, due to the sheer quantity of files. Please review them and fix these issues, before I can continue.

File: src/main/java/org/eclipse/tractusx/ssi/lib/exception/resolver/DidWebException.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ * *******************************************************************************
+ */
+
+package org.eclipse.tractusx.ssi.lib.exception.resolver;
+
+import org.eclipse.tractusx.ssi.lib.exception.SSIException;
+
+public class DidWebException extends SSIException {
+
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/exception/resolver/DidWebException.java:27]: Missing `serialVersionUID = 1L`

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/did/DidParser.java [hunk 6]
```diff
@@ -30,12 +30,6 @@
 
 /** The type Did parser. */
 public class DidParser {
-  /**
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/model/did/DidParser.java:33]: Restore comments for Maven Central compatibility.

File: src/main/java/org/eclipse/tractusx/ssi/lib/proof/transform/LinkedDataTransformer.java [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /*
  * ******************************************************************************
- * Copyright (c) 2021,2024 Contributors to the Eclipse Foundation
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/proof/transform/LinkedDataTransformer.java:3]: Copyright year is wrong.

File: src/main/java/org/eclipse/tractusx/ssi/lib/proof/types/jws/JWSProofVerifier.java
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/proof/types/jws/JWSProofVerifier.java:1]: Missing copyright header update.

File: src/test/java/org/eclipse/tractusx/ssi/lib/proof/LinkedDataProofValidationComponentTest.java
Comment (borisrizov-zf) [inline src/test/java/org/eclipse/tractusx/ssi/lib/proof/LinkedDataProofValidationComponentTest.java:1]: Missing copyright header update

File: src/test/java/org/eclipse/tractusx/ssi/lib/util/identity/TestIdentityFactory.java [hunk 8]
```diff
@@ -43,11 +43,6 @@
 /** The type Test identity factory. */
 public class TestIdentityFactory {
 
-  /**
```
Comment (borisrizov-zf) [inline src/test/java/org/eclipse/tractusx/ssi/lib/util/identity/TestIdentityFactory.java:46]: Restore comment.

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

**Comment by borisrizov-zf** [comment]
@koptan can you point me in the right direction about the following:

verification method can be either a multibase string or a map, but I cannot see where transformation is taking place. The interface which binds these seems to be missing the option to easily extend to something like `publicKeyJWK` with an EC cipher. Am I missing something?

**Comment by koptan** [comment]
@borisrizov-zf transformation is happening in TransformedLinkedData, we accept any JsonLD object as parameter.

File: src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtFactory.java [hunk 9]
```diff
@@ -116,10 +116,13 @@ private static SignedJWT createSignedES256Jwt(
...
-          new JWSHeader(
-              algorithm, type, null, null, null, null, null, null, null, null, issuer, true, null,
-              null);
+
+      JWSHeader.Builder jwsHeaderBuilder = new JWSHeader.Builder(algorithm);
+      jwsHeaderBuilder.type(type);
+      jwsHeaderBuilder.keyID(issuer);
+      jwsHeaderBuilder.base64URLEncodePayload(true);
+
+      var header = jwsHeaderBuilder.build();
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtFactory.java:125]:
```suggestion
      JWSHeader header = new JWSHeader.Builder(algorithm)
              .type(type)
              .keyID(issuer)
              .base64URLEncodePayload(true)
              .build();
```

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

**Review by borisrizov-zf — APPROVED** [review]

**Comment by borisrizov-zf** [comment]
@koptan Hi, the branch needs rebasing, you're out of sync with main.

**Pull request merged**